providers/implementations/ciphers/cipher_aes_ocb.c

```diff
@@ -401,11 +401,12 @@ static int aes_ocb_get_ctx_params(void *vctx, OSSL_PARAM params[])
 
     p = OSSL_PARAM_locate(params, OSSL_CIPHER_PARAM_IV);
     if (p != NULL) {
-        if (ctx->base.ivlen != p->data_size) {
+        if (ctx->base.ivlen > p->data_size) {
```

Comment:
Isn't the data_size the size of the pointer in the case of octet_ptr?

Reply:
I confess I'm not 100% clear on that; in the case I link to (`EVP_CIPHER_CTX_iv_noconst`) we pass the size of the `iv[]` array, at least.

Reply:
I don't see anything in either OSSL_PARAM_BLD.pod or OSSL_PARAM_int.pod that suggests that the size should be the size of the pointer.
The implementation of `OSSL_PARAM_BLD_push_octet_ptr()`, however, does:
https://github.com/openssl/openssl/blob/00da0f69890874feaa555fafb99b967b861e9118/crypto/param_build.c#L287-L302
Could you say a bit more about why you believe the pointer size is correct? In some sense it seems like the value is unused, since we're not providing a buffer, and are instead getting a live pointer to someone else's allocated storage of the proper length.

Reply:
In OSSL_PARAM.doc, there's this section (rendered here):

>   _data_size_ must be set to the size of the data, not the size of the
>   pointer to the data.  If this is used in a parameter request,
>   _data_size_ is not relevant.  However, the _responder_ will set
>   _return_size_ to the size of the data.

Reply:
I recall that we discussed what the data size for a indirect data pointer (which is what a `OSSL_PARAM_OCTET_PTR` _data_ really is) should be the pointer size or the size of the value being (indirectly) pointed at.  @mattcaswell may simply misremember how we concluded that discussion.

Reply:
So, Richard, it seems that you are saying that Matt was incorrect to request changes here?
(Does anyone care to push the 'approve' button? ;)

Reply:
Or is the claim that I should change it to only check the length in the octet-string case?

Reply:
@levitte some clarification would be appreciated

Reply:
I can only repeat what I just pointed out, that `data_size` should be set to the size of the data, i.e. the size of the storage that the pointer that `data` points at is pointing at.

(verbal indirect reference... it just had to be done :wink:)